src/deltacode/__init__.py

```diff
@@ -112,111 +137,78 @@ def determine_delta(self):
-        new_visited, old_visited = 0, 0
 
-        # perform the deltas
-        for path, new_files in new_index.items():
-            for new_file in new_files:
-
-                if new_file.type != 'file':
-                    continue
+        new_visited = 0
+        old_visited = 0
+        old_files_sha1_considered_in_deltas = dict()
 
+        for new_resource in self.codebase1.walk():
+            if new_resource.is_file:
                 new_visited += 1
-
-                try:
-                    delta_old_files = old_index[path]
-                except KeyError:
-                    delta = Delta(100, new_file, None)
-                    delta.status = 'added'
+                path_new = self.new_files_original_path[new_resource.path]
+                ADDED = True
+                for old_resource in self.codebase2.walk():
+                    if old_resource.is_file and not old_resource.sha1 in old_files_sha1_considered_in_deltas.keys():
+                        old_visited += 1
+                        path_old = self.old_files_original_path[old_resource.path]
+                        if path_new == path_old:
+                            ADDED = False
+                            if new_resource.sha1 == old_resource.sha1:
+                                old_files_sha1_considered_in_deltas[old_resource.sha1] = 1
+                                self.stats.num_unmodified += 1
+                                new_resource.path = path_new
+                                old_resource.path = path_old
+                                delta = Delta(0, new_resource, old_resource)
+                                delta.status = "unmodified"
+                                self.deltas.append(delta)
+                                break
+                            else:
+                                old_files_sha1_considered_in_deltas[old_resource.sha1] = 1
+                                self.stats.num_modified += 1
+                                new_resource.path = path_new
+                                old_resource.path = path_old
+                                delta = Delta(20, new_resource, old_resource)
+                                delta.status = "modified"
+                                self.deltas.append(delta)
+                                break
+                        else:
+                            if new_resource.sha1 == old_resource.sha1:
+                                old_files_sha1_considered_in_deltas[old_resource.sha1] = 1
+                                ADDED = False
+                                self.stats.num_moved += 1
+                                new_resource.path = path_new
+                                old_resource.path = path_old                                
+                                delta = Delta(0, new_resource, old_resource)
+                                delta.status = "moved"
+                                self.deltas.append(delta)
```

Comment:
Its probably cleaner to have functions to create each type of Delta object (Added, modified, etc)

That way we do not have 6+ lines of code after each conditional.

Reply:
Yes it can be done.
It is a just a basic model which I did to show the working of the BFS traversal of the two codebases.

Reply:
Changed in commit https://github.com/nexB/deltacode/pull/167/commits/e6a6cbdd0eb3d850e21f46be658d24a8dfa6bca8